File: velolabs_repos/lattis_ios/LattisCore/Model/Sources/Model/Subscription.swift
//
//  Subscription.swift
//  LattisCore
//
//  Created by Ravil Khusainov on 03.08.2020.
//  Copyright © 2020 Lattis inc. All rights reserved.
//

import Foundation


public struct Subscription: Codable {
    public let id: Int
    public let activatedAt: Date
    public let deactivatedAt: Date?
    public let periodStart: Date
    public let periodEnd: Date
    public let membership: Membership
}

public extension Subscription {
    enum CodingKeys: String, CodingKey {
        case id = "membershipSubscriptionId"
        case activatedAt = "activationDate"
        case deactivatedAt = "deactivationDate"
        case periodStart
        case periodEnd
        case membership = "fleetMembership"
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/models/Subscription.kt
package com.lattis.domain.models

import java.io.Serializable

data class Subscription(
    var membership_subscription_id:Int?=null,
    var fleet_membership_id:Int?=null,
    var user_id:Int?=null,
    var activation_date:String?=null,
    var deactivation_date:String?=null,
    var period_start:String?=null,
    var period_end:String?=null,
    var fleet_membership: Membership?=null
):Serializable